various: remove by-name overrides

qweered opened the pull request:
Continuation of #469771, resurrection of #453948

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

wolfgangwalther committed:
by-name/various: move overrides to package.nix

qweered commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 474456`
Commit: [`420441c00240f3fc7ed7b20932ae95509b4fc72e`](https://github.com/NixOS/nixpkgs/commit/420441c00240f3fc7ed7b20932ae95509b4fc72e) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/420441c00240f3fc7ed7b20932ae95509b4fc72e..pull/474456/head))
Merge: [`21a648285005177c7c348cae7a65dd4f988dd0f6`](https://github.com/NixOS/nixpkgs/commit/21a648285005177c7c348cae7a65dd4f988dd0f6)

Logs: https://github.com/qweered/nixpkgs-review-gha/actions/runs/20531911443


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>devilutionx</li>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
    <li>saunafs</li>
    <li>saunafs.dev</li>
    <li>saunafs.man</li>
    <li>smpq</li>
    <li>tlaps</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>devilutionx</li>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
    <li>saunafs</li>
    <li>saunafs.dev</li>
    <li>saunafs.man</li>
    <li>smpq</li>
    <li>tlaps</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>tlaps</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>isabelle</summary>
<pre>no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Building isabelle setup
Building HOL heap
Started at Sat Dec 27 00:47:22 GMT 2025 (polyml-for-isabelle-2025_x86_64-darwin on iad20-fj923-6d012934-81ee-499f-a2f6-83a838fac294-F64FF27E8B9A)
ISABELLE_TOOL_JAVA_OPTIONS=&quot;-Djava.awt.headless=true -Xms512m -Xmx4g -Xss16m&quot;
ISABELLE_BUILD_OPTIONS=&quot;&quot;

ML_PLATFORM=&quot;x86_64-darwin&quot;
ML_HOME=&quot;/nix/store/bx6mg1dfidsgd8y337xsla20wyzjv6bx-polyml-for-isabelle-2025/bin&quot;
ML_SYSTEM=&quot;polyml-for-isabelle-2025&quot;
ML_OPTIONS=&quot;--minheap 1000&quot;

2025-12-27 00:47:26.162 java[12254:58066] CoreText: System LastResort not available, using built-in copy.
Session Pure/Pure
Session Misc/Tools
Session HOL/HOL (main)
Building Pure ...
*** I/O error: Operation not permitted</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>tlaps</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>isabelle</summary>
<pre>no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Building isabelle setup
Building HOL heap
Started at Sat Dec 27 00:42:16 GMT 2025 (polyml-for-isabelle-2025_aarch64-darwin on Mac-1766795626772)
ISABELLE_TOOL_JAVA_OPTIONS=&quot;-Djava.awt.headless=true -Xms512m -Xmx4g -Xss16m&quot;
ISABELLE_BUILD_OPTIONS=&quot;&quot;

ML_PLATFORM=&quot;aarch64-darwin&quot;
ML_HOME=&quot;/nix/store/1krg79r5pvf1mhpqcj867sj2lcchbpyp-polyml-for-isabelle-2025/bin&quot;
ML_SYSTEM=&quot;polyml-for-isabelle-2025&quot;
ML_OPTIONS=&quot;--minheap 1000&quot;

2025-12-27 00:42:16.686 java[9801:46422] CoreText: System LastResort not available, using built-in copy.
Session Pure/Pure
Session Misc/Tools
Session HOL/HOL (main)
Building Pure ...
*** I/O error: Operation not permitted</pre>
</details>
</details>

gepbird commented on pkgs/by-name/ra/rabbitmq-server/package.nix:
```diff
@@ -39,6 +40,8 @@ let
     ]
   );
+
+  beamPackages = beam27Packages.extend (self: super: { elixir = elixir_1_17; });
```
This was using `elixir_1_18`, if this change is intentional, I think it should go in a new commit.

gepbird commented on pkgs/by-name/go/gopro-tool/package.nix:
```diff
@@ -5,9 +5,16 @@
 }:
 
+let
+  vlc' = vlc.overrideAttrs (old: {
```
Outside of the scope of this PR, but we shouldn't really use `overrideAttrs` inside nixpkgs: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#overrideattrs-and-overridepythonattrs.

gepbird commented on pkgs/by-name/gl/glib/package.nix:
This diff is a not trivial, maybe split the actual changes and nixfmt into separate commits?

gepbird reviewed (commented):
Thanks for picking this up!

I went through the whole diff except glib as I'm out of time now, mostly looks good. Big thanks for splitting up the commits per package, those were easier to review than the `various:` commits. There were a few unrelated changes like `rec` -> `finalAttrs`, `with libs.platforms` removal or removing outdated overrides as suggested previously which is fine, but I think you could keep the scope a little narrower next time.

emilazy commented:
AFAICT, this PR does not address the fundamental issues caused by current `by-name` limitations around sometimes producing unusable override interfaces (e.g. in the case of conditional `stdenv`s) that led to @wolfgangwalther’s original PR being abandoned, right?

> * I feel like we really should avoid the `{ ... }@args` pattern. It's complex, difficult to understand for new maintainers, and there's really no good reason to have it when we can just have `luajit` in the list of function inputs.

Avoiding it produces the worse outcome where you have a load‐bearing `foo'` but can still refer to the non‐overridden `foo` later in the package, as well as resulting in more code churn.

qweered commented:
for ```{ ... }@args``` or ```foo'``` problem we have rfc https://github.com/NixOS/rfcs/pull/192 and implementation https://github.com/NixOS/nixpkgs/pull/464396

qweered committed:
glib: move glib-untested overrides into package.nix

This has the additional benefit, that the tests will run with
`finalPackage`, i.e. taking overrides into account.

qweered commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 474456`
Commit: [`fd6f219ec7ab9d1751fb071d963a9a106bd5ba04`](https://github.com/NixOS/nixpkgs/commit/fd6f219ec7ab9d1751fb071d963a9a106bd5ba04) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/fd6f219ec7ab9d1751fb071d963a9a106bd5ba04..pull/474456/head))
Merge: [`37b0b836ef315fb2fd0a5384ed1e22031086c7e2`](https://github.com/NixOS/nixpkgs/commit/37b0b836ef315fb2fd0a5384ed1e22031086c7e2)

Logs: https://github.com/qweered/nixpkgs-review-gha/actions/runs/20904360298


---
### `x86_64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>tlaps</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>devilutionx</li>
    <li>saunafs</li>
    <li>saunafs.dev</li>
    <li>saunafs.man</li>
    <li>smpq</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>tlaps</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>devilutionx</li>
    <li>saunafs</li>
    <li>saunafs.dev</li>
    <li>saunafs.man</li>
    <li>smpq</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>tlaps</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>tlaps</li>
  </ul>
</details>

sempiternal-aurora commented:
> isabelle is failing previously because of vampire, tlaps depeds on isabelle. I think its good to merge

For Isabelle, pr #474227 will fix the build, but also remove the by-name override.

qweered commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 474456`
Commit: [`c2592faa567a47401854c67ee00a53f06350be93`](https://github.com/NixOS/nixpkgs/commit/c2592faa567a47401854c67ee00a53f06350be93) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/c2592faa567a47401854c67ee00a53f06350be93..pull/474456/head))
Merge: [`46b2af46774f0170506f6f62c81b96eb54054458`](https://github.com/NixOS/nixpkgs/commit/46b2af46774f0170506f6f62c81b96eb54054458)

Logs: https://github.com/qweered/nixpkgs-review-gha/actions/runs/20991281077


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>devilutionx</li>
    <li>saunafs</li>
    <li>saunafs.dev</li>
    <li>saunafs.man</li>
    <li>smpq</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>devilutionx</li>
    <li>saunafs</li>
    <li>saunafs.dev</li>
    <li>saunafs.man</li>
    <li>smpq</li>
  </ul>
</details>

qweered commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 474456`
Commit: [`8025622e8247ac1177601be91aeb984629593109`](https://github.com/NixOS/nixpkgs/commit/8025622e8247ac1177601be91aeb984629593109) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/8025622e8247ac1177601be91aeb984629593109..pull/474456/head))
Merge: [`14bc6a6a21ec77ec65ea581ba6dccb99f06022ef`](https://github.com/NixOS/nixpkgs/commit/14bc6a6a21ec77ec65ea581ba6dccb99f06022ef)

Logs: https://github.com/qweered/nixpkgs-review-gha/actions/runs/21226328872


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>devilutionx</li>
    <li>saunafs</li>
    <li>saunafs.dev</li>
    <li>saunafs.man</li>
    <li>smpq</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>devilutionx</li>
    <li>saunafs</li>
    <li>saunafs.dev</li>
    <li>saunafs.man</li>
    <li>smpq</li>
  </ul>
</details>

mdaniels5757 commented:
Rebased, which dropped the following commits: bc61ec461bd0, 439778ed0473, 5a635e45b538, a122428fe0a8, 8561b049ef43, 2e6967bda095

mdaniels5757 commented:
@qweered I guess the only things that haven't been split are c292e7961e12ef92e9fc1b99f3f7b081c8edb263 and b3ce16e7405a09f6b79135e4ae8d93edb0324dde.

I think c292e7961e12ef92e9fc1b99f3f7b081c8edb263 should probably be split up to be one PR for package. That way, each packages' maintainers can decide how they'd like to remove their all-packages.nix override, and this doesn't block the other packages. Would you mind splitting this one up and making those PRs? (In case you didn't already know about it: `git add -p` is your friend :) )

b3ce16e7405a09f6b79135e4ae8d93edb0324dde should be its own PR too, I think. That one is easy, but I don't want to make the PR without reading the diff, and I don't want to read all that... :)

gepbird commented:
> [b3ce16e](https://github.com/NixOS/nixpkgs/commit/b3ce16e7405a09f6b79135e4ae8d93edb0324dde) should be its own PR too, I think. That one is easy, but I don't want to make the PR without reading the diff, and I don't want to read all that... :)

That diff isn't that scary after you hide whitespace (I learned it from @qweered in https://github.com/NixOS/nixpkgs/pull/474456#discussion_r2679950039).

qweered commented:
every commit from this brach has been splitted into separate pr
